url2b4a2d655b_Yehonal_unisa_prog2_jdm_tgz-pJ8-com_jdeals_tests_TestUserBuyingJ8/src/com/jdeals/main/helper/storage/FileSerialization.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (C) 2007 - 2014 Hyperweb2 All rights reserved.
+ * GNU General Public License version 3; see www.hyperweb2.com/terms/
+ */
+package com.jdeals.main.helper.storage;
+
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.io.Serializable;
+import java.util.ArrayList;
+
+/**
+ * The Class FileSerialization.
+ *
+ * @param <T> the generic type
+ */
+public class FileSerialization<T extends Serializable> implements
+        DataBaseInterface<T> {
+
+    /* (non-Javadoc)
+     * @see com.jdeals.main.helper.storage.DataBaseInterface#storeList(java.util.ArrayList, java.lang.String)
+     */
+    @Override
+    public void storeList(ArrayList<T> list, String filePath) {
+        try {
+            FileOutputStream fileOut = new FileOutputStream(filePath);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *fileOut*. The resource is referenced at lines 30, 32. The resource is closed at lines 34, 35. There are other execution paths that do not contain closure statements, for example, when ObjectOutputStream.writeObject throws an exception. Ensure *fileOut* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).